packages/nextjs/src/app-router/server/ClerkProvider.tsx

```diff
@@ -6,7 +6,9 @@ import { mergeNextClerkPropsWithEnv } from '../../utils/mergeNextClerkPropsWithE
 import { ClientClerkProvider } from '../client/ClerkProvider';
 import { initialState } from './auth';
 
-export function ClerkProvider(props: Without<NextClerkProviderProps, '__unstable_invokeMiddlewareOnAuthStateChange'>) {
+export function ClerkProvider(
+  props: Without<NextClerkProviderProps, '__unstable_invokeMiddlewareOnAuthStateChange' | 'signInUrl' | 'signUpUrl'>,
```

Comment:
I've deliberately removed `signInUrl` and `signUpUrl` from the allowed props of `ClerkProvider` on the server. 

IMO limiting it only to be passed via the `clerkMiddleware` would cause less confusion, one single source of truth for those options to be used by server-side helpers. 

If we propagate within `ClerkProvider` on the server to the request, then there's the question of which would would take precedence: Middleware options or provider options?

Reply:
This makes lots of sense to me, but unfortunately, we can't do it at the moment as it's a breaking change :( 

I agree that the options of `clerkMiddleware` should take precedence over the ones defined on `ClerkProvider`.

What would happen if we keep both right now and make sure that the `clerkMiddleware` ones override all else?

Reply:
How would these options propagate to components that are not server rendered? We would need to pass the options from middleware to the client (middlware -> server -> client). My recommendation would be to not attempt that, scope it to the properties needed by our server-side helpers for now, and document the constraints clearly.

What do y'all think?

Reply:
> scope it to the properties needed by our server-side helpers for now, and document the constraints clearly

+1 on that! I'll start documenting the constraints for each helper/component.